client/my-sites/add-ons/hooks/use-add-on-display-cost.ts

```diff
@@ -6,10 +6,10 @@ import useAddOnPrices from './use-add-on-prices';
 const useAddOnDisplayCost = ( productSlug: string, quantity?: number ) => {
 	const translate = useTranslate();
 	const prices = useAddOnPrices( productSlug, quantity );
-	const formattedCost = prices?.formattedMonthlyPrice || '';
 
 	return useSelector( ( state ) => {
 		const product = getProductBySlug( state, productSlug );
+		const formattedCost = prices?.formattedMonthlyPrice || '';
```

Comment:
Is this intended? I'm asking since it doesn't need anything inside the `return useSelector( ...` block to compute, so from the look of it I'd expect it should go right under `const prices = useAddOnPrices ...` line above.

Reply:
I'm not sure - it seems intended, but also won't harm to move it back. I think intention was to move it closer to the scope that it was needed and just keep those hook calls in the outer

Reply:
Well, actually, i think this should have been _more_ intended than i explained :-) 

When we think of micro-optimising, then it feels like unnecessary assignment on every time function gets called, if sitting in the outer scope like that. The callback should get optimised/memoised, may only run once (in case of useMemo for instance), etc.

Micro, but the little things add up. Now I'm not sure if anything else happens behind the scenes and my tbinming isn't technically valid - but I am assuming a combination of "worst-case" assignments & closure, are worse that a single (potentially) inner-scope variable assignment 🤔

Reply:
I reverted, but might attempt to sneak the change back in some other existing PR